### PR TITLE
Support python 3.8-3.12, and pypy

### LINKED ISSUE #417: Update CI python versions to 3.8-3.12
I think we can safely drop 3.7 and 3.8 at this point, and add 3.12.

### PR DESCRIPTION
**Pull request by almarklein**
Closes #417

### CODE REVIEW

**Comment by almarklein** [comment]
@Korijn did you change the branch protection rules already? I was looking to update them, but it was magically fixed 😺

**Comment by Korijn** [comment]
Yes! I wanted to help. Haha.

**Comment by almarklein** [comment]
Thanks 🙏

**Review by Korijn — APPROVED** [review]
LGTM. Why does pypy need manual gc?

**Comment by almarklein** [comment]
Pypy is more conservative wrt releasing memory and often needs more iterations to let loose all refs. I've seen this in other projects too. It feels a bit "unscientific" to just throw more `gc.collect()`'s in there, but it works!

File: tests_mem/testutils.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def _determine_can_use_pyside6():
 can_use_glfw = _determine_can_use_glfw()
 can_use_pyside6 = _determine_can_use_pyside6()
 is_ci = bool(os.getenv("CI", None))
+is_pypy = "__pypy__" in sys.builtin_module_names
```
Comment (hmaarrfk) [inline tests_mem/testutils.py:60]: FYI, numpy uses:
```
IS_PYPY = sys.implementation.name == 'pypy'
```
https://github.com/numpy/numpy/blob/83112cd66b2b26c41aac7204860050e5ad7fed12/numpy/_core/_internal.py#L19

Comment (almarklein) [reply tests_mem/testutils.py:60]: #419

Comment (almarklein) [reply tests_mem/testutils.py:60]: Thanks!

**Pull request merged**